scvi/model/base/_demixin.py

```diff
@@ -0,0 +1,453 @@
+from torch.distributions import Categorical, Normal
+
+from scvi._compat import Literal
+from scvi._docs import doc_differential_expression
+from scvi._utils import _doc_params
+from scvi.model._utils import (
+    _get_batch_code_from_category,
+    _get_var_names_from_setup_anndata,
+    scrna_raw_counts_properties,
+)
+from scvi.model.base._utils import _de_core
+
+logger = logging.getLogger(__name__)
+Number = Union[int, float]
+
+
+class DEMixin:
+    """
+    DE module relying on Importance-sampling.
+
+    Mixin for using
+    importance-weighted DE content.
+    This however requires some additional structure on the
+    module's (e.g., VAE) methods and associate signatures
+    """
+
+    @_doc_params(
+        doc_differential_expression=doc_differential_expression,
+    )
+    def lvm_de(
+        self,
+        adata: Optional[AnnData] = None,
+        groupby: Optional[str] = None,
+        group1: Optional[Iterable[str]] = None,
+        group2: Optional[str] = None,
+        idx1: Optional[Union[Sequence[int], Sequence[bool]]] = None,
+        idx2: Optional[Union[Sequence[int], Sequence[bool]]] = None,
+        mode: Literal["vanilla", "change"] = "change",
+        delta: float = None,
+        batch_size: Optional[int] = None,
+        all_stats: bool = True,
+        batch_correction: bool = False,
+        batchid1: Optional[Iterable[str]] = None,
+        batchid2: Optional[Iterable[str]] = None,
+        fdr_target: float = 0.05,
+        silent: bool = False,
+        pseudocounts: float = None,
+        fn_kwargs: Optional[dict] = None,
+        **kwargs,
+    ) -> pd.DataFrame:
+        r"""
+        A unified method for differential expression analysis using lvm-DE.
+
+        Parameters
+        ----------
+        {doc_differential_expression}
+        **kwargs
+            Keyword args for :func:`scvi.utils.DifferentialComputation.get_bayes_factors`
+
+        Returns
+        -------
+        Differential expression DataFrame.
+        """
+        adata = self._validate_anndata(adata)
+        fn_kwargs = dict() if fn_kwargs is None else fn_kwargs
+        col_names = _get_var_names_from_setup_anndata(adata)
+        model_fn = partial(
+            self.get_population_expression,
+            return_numpy=True,
+            batch_size=batch_size,
+            **fn_kwargs,
+        )
+        result = _de_core(
+            adata,
+            model_fn,
+            groupby,
+            group1,
+            group2,
+            idx1,
+            idx2,
+            all_stats,
+            scrna_raw_counts_properties,
+            col_names,
+            mode,
+            batchid1,
+            batchid2,
+            delta,
+            batch_correction,
+            fdr_target,
+            silent,
+            pseudocounts=pseudocounts,
+            use_permutation=True,
+            **kwargs,
+        )
+
+        return result
+
+    @torch.no_grad()
+    def get_population_expression(
```

Comment:
is this the best name for it? Is this something you expect users to call independently of DE?

Reply:
`get_subpopulation_expression`